lib/rules/unit-blacklist/index.js

```diff
@@ -31,18 +40,64 @@ const rule = function(blacklistInput, options) {
         optional: true,
         actual: options,
         possible: {
-          ignoreProperties: validateObjectWithStringArrayProps
+          ignoreProperties: validateObjectWithStringArrayProps,
+          ignoreMediaFeatureNames: validateObjectWithStringArrayProps
         }
       }
     );
     if (!validOptions) {
       return;
     }
 
-    function check(node, value, getIndex) {
+    function checkMedia(node, value, getIndex) {
+      value = value.replace(/\*/g, ",");
+
+      mediaParser(node.params).walk(/^media-feature$/i, mediaFeatureNode => {
+        const mediaName = getMediaFeatureName(mediaFeatureNode),
+          parentValue = mediaFeatureNode.parent.value;
+
+        valueParser(value).walk(function(valueNode) {
+          // Ignore all non-word valueNode and
+          // the values not included in the parentValue string
+          if (
+            valueNode.type !== "word" ||
+            !parentValue.includes(valueNode.value)
+          ) {
+            return;
+          }
+
+          const unit = getUnitFromValueNode(valueNode);
```

Comment:
From here to the end of the function is almost a mirror image of the same thing in walk.decls. I think it should be abstracted.

Reply:
Done in my last commit too b6effdf